tmk_core/protocol/vusb/vusb.c

```diff
@@ -101,6 +101,45 @@ static uint8_t           kbuf_tail = 0;
 
 static report_keyboard_t keyboard_report_sent;
 
+static void send_report_fragment(uint8_t endpoint, void *data, size_t size) {
+    switch (endpoint) {
+        case 1:
+            while (!usbInterruptIsReady()) {
+                usbPoll();
+            }
+            usbSetInterrupt(data, size);
+            break;
+        case USB_CFG_EP3_NUMBER:
+            while (!usbInterruptIsReady3()) {
+                usbPoll();
+            }
+            usbSetInterrupt3(data, size);
+            break;
+        case USB_CFG_EP4_NUMBER:
+            while (!usbInterruptIsReady4()) {
+                usbPoll();
+            }
+            usbSetInterrupt4(data, size);
+            break;
+    }
```

Comment:
The infinite timeout as implemented here may not be appropriate at least for the console endpoint (it may result in a hang if the debug console is enabled, but there is no listening program running on the host).  Other endpoints may cause problems too (e.g., if some report gets sent to the shared endpoint while the host is using the boot protocol and does not poll that endpoint).